pkg/resourceoverrides/cloudids_overrides.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourceoverrides
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/k8s"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func GetCloudIDSEndpointResourceOverrides() ResourceOverrides {
+	return ResourceOverrides{
+		Kind: "CloudIDSEndpoint",
+		Overrides: []ResourceOverride{
+			mapEndpointIpToEndpointIP(),
+		},
+	}
+}
+
+func mapEndpointIpToEndpointIP() ResourceOverride {
+	return ResourceOverride{
+		CRDDecorate: func(crd *apiextensions.CustomResourceDefinition) error {
+			schema := k8s.GetOpenAPIV3SchemaFromCRD(crd)
+			status, ok := schema.Properties["status"]
+			if !ok {
+				return fmt.Errorf("status field not found")
+			}
+			observedState, ok := status.Properties["observedState"]
+			if !ok {
+				return fmt.Errorf("observedState field not found")
+			}
+			endpointIp, ok := observedState.Properties["endpointIp"]
+			if !ok {
+				return fmt.Errorf("endpointIp field not found")
+			}
+
+			// Rename endpointIp to endpointIP
+			observedState.Properties["endpointIP"] = endpointIp
+			delete(observedState.Properties, "endpointIp")
+
+			return nil
+		},
+		PostActuationTransform: func(original, reconciled *k8s.Resource, tfState *terraform.InstanceState, dclState *unstructured.Unstructured) error {
+			reconciled.Status["endpointIP"] = reconciled.Status["endpointIp"]
```

Comment:
Should the logic be:

If `status.endpointIp` exists, add field `status.endpointIP` and remove field `status.endpointIp`?

Reply:
Yes initially the PR was a wip and I got a chance to polish it up now